pkg/syncer/apis/edge/v1alpha1/edge-syncconfig.go

```diff
@@ -0,0 +1,78 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster,shortName=esc
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type EdgeSyncConfig struct {
+	/*
+	 */
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   EdgeSyncConfigSpec   `json:"spec,omitempty"`
+	Status EdgeSyncConfigStatus `json:"status,omitempty"`
+}
+
+// EdgeSyncConfigSpec defines the desired state of EdgeSyncConfig
+type EdgeSyncConfigSpec struct {
+	// Downsncyed resource list
+	DownSyncedResources []EdgeSyncConfigResource `json:"downSyncedResources,omitempty"`
+
+	// Upsncyed resource list
+	UpSyncedResources []EdgeSyncConfigResource `json:"upSyncedResources,omitempty"`
+
+	// Conversions
+	Conversions []EdgeSynConversion `json:"conversions,omitempty"`
+}
+
+// Resource specifies down/up synced resource with exact GVK and name (and namespace if not cluster scoped resource)
+type EdgeSyncConfigResource struct {
+	// Kind of down/up synced resource
+	Kind string `json:"kind,omitempty"`
```

Comment:
The implementation of the syncer would be simpler if this field held the resource rather than the kind.  For example, "deployments" rather than "Deployment".  That would remove the need to go through "rest mapping" to get from kind to resource.

Reply:
Yes, I agree with. Actually, I had to write the code to go through rest mapping. Though, as a user who creates EdgeSyncConfig, I am wondering if "resource" is always represented as lower case of "kind" with "s" appended (; `strings.ToLower(kind) + "s"`). Actually, I can find "kind" in k8s yaml manifest but not find "resource". Maybe it's responsible of a user who creates EdgeSyncConfig with correct "resource" by somehow finding it from somewhere.